core/src/main/java/org/apache/calcite/sql/SqlSplittableAggFunction.java

```diff
@@ -173,11 +173,11 @@ public RexNode singleton(RexBuilder rexBuilder, RelDataType inputRowType,
       final RexNode predicate =
           RexUtil.composeConjunction(rexBuilder, predicates, true);
       if (predicate == null) {
-        return rexBuilder.makeExactLiteral(BigDecimal.ONE);
+        return rexBuilder.makeBigintLiteral(BigDecimal.ONE);
```

Comment:
Let's not assume that COUNT is BIGINT. (In some type systems it might be INT.) Use AggregateCall.type?